addons/web/static/src/js/view_form.js

```diff
@@ -6310,6 +6310,44 @@ instance.web.form.StatInfo = instance.web.form.AbstractField.extend({
 
 });
 
+/**
+    This widget is intended to be used on boolean fields. It toggles a button
+    siwtching between a green bullet / gray bullet.
+*/
+instance.web.form.FieldToggleBoolean = instance.web.form.AbstractField.extend({
+    init: function (field_manager, node) {
+        this._super.apply(this, arguments);
+        this.icon = "";
+    },
+    render_value: function () {
+        var self = this;
+        self.icon = self.get_value() ? 'gtk-yes' : 'gtk-normal';
+        self.$el.html(QWeb.render("toggle_button", {'widget': self, 'prefix': instance.session.prefix,}));
+        self.$el.find('button').on('click', self.set_toggle_button.bind(self));
+    },
```

Comment:
No need of render value, those thing can be done from template, bind set_toggle_button in start method.
Template *toggle_button*, button is used in list if we change for list view, it will also affects here, so make new template. @msa-odoo